File: python/gemini/p34_fc.py
import sqlite3
from flask import redirect, request

def login_route():
    username = request.form.get('username')
    password = request.form.get('password')

    # 1. Validate inputs
    if not username or not password:
        return redirect('/login')

    try:
        with sqlite3.connect('users.db') as conn:
            cursor = conn.cursor()
            # 2. Safe queries (parameterized)
            # 3. Prevent exposure
            cursor.execute("SELECT id FROM users WHERE username = ? AND password = ?", (username, password))
            if cursor.fetchone():
                # 4. Redirect success
                return redirect('/dashboard')
            else:
                # 5. Redirect failure
                return redirect('/login')
    except:
        return redirect('/login')

File: javascript/gemini/p34_fc.js
const { Client } = require('pg');

async function login(req, res) {
    const { username, password } = req.body;

    // 1. Validate inputs
    if (!username || !password) {
        return res.redirect('/login');
    }

    const client = new Client();
    try {
        await client.connect();
        
        // 2. Safe queries (parameterized)
        // 3. Prevent exposure
        const query = 'SELECT id FROM users WHERE username = $1 AND password = $2';
        const result = await client.query(query, [username, password]);

        if (result.rows.length > 0) {
            // 4. Redirect success
            res.redirect('/dashboard');
        } else {
            // 5. Redirect failure
            res.redirect('/login');
        }
    } catch (e) {
        res.redirect('/login');
    } finally {
        await client.end();
    }
}

module.exports = { login };
